packages/swagger-swr/src/builders/QueryBuilder.ts

```diff
@@ -24,22 +24,26 @@ export class QueryBuilder extends OasBuilder<Config> {
 
     const name = camelCase(`${operation.getOperationId()}QueryOptions`)
 
-    const generics = [`TData = ${schemas.response.name}`, `TError = ${errors.map((error) => error.name).join(' | ') || 'unknown'}`]
+    const generics = [
+      `TData = ${schemas.response.name}`,
+      `TError = ${errors.map((error) => error.name).join(' | ') || 'unknown'}`,
+    ]
     const clientGenerics = ['TData', 'TError']
     const queryGenerics = [dataReturnType === 'data' ? 'TData' : 'ResponseConfig<TData>', 'TError']
     const params = createFunctionParams([
-      ...getDataParams(schemas.pathParams, { typed: true }),
+      ...getDataParams(schemas.pathParams, { typed: true })
+        .map(param => ({ ...param, required: false })),
       {
         name: 'params',
         type: schemas.queryParams?.name,
         enabled: !!schemas.queryParams?.name,
-        required: !!schemas.queryParams?.schema.required?.length,
+        required: false,
```

Comment:
@stijnvanhulle It's already been merged, but I'll answer it. I think it's better to allow React to not pass a value when it doesn't immediately know what value it needs for every parameter, rather than trying to pass meaningless fake values.

For example

```js
const token = useToken()
const userInfo = useGetUserInfo(token ? { token } : undefined, { shouldFetch: !!token })
```

When parameters are required, you should do it like below, but it is inconvenient to use `|| ''` for string type.

```js
const userInfo = useGetUserInfo({ token: token || '' }, { shouldFetch: !!token })
```